### ISSUE TITLE
Add Option to Delete Specific Session Data

### ISSUE THREAD

**Issue by Kntntn** [umami-software/umami#2940]
### Describe the feature or enhancement

It would be helpful to have a feature that allows users to delete specific session data. This would be particularly useful for instances where accidental visits to the website by the site owner or team members skew the analytics. A simple button to remove selected sessions from the database would suffice.

Is there already an existing feature or workaround that allows for the deletion of specific session data without directly accessing the database? If not, would this make sense to implement?

**Comment by cdog2000** [comment]
This would be a very helpful feature and I use it regularly in Matomo while working and doing tests on my website on different devices. I really hope it will be implemented in the near future. 🙏🏻

**Comment by GuillaumeChk** [comment]
I agree, same problem. I thinked about a **workaround** to delete manually the data of a specific visitor. I can easily see which visitor is someone from my team testing the site : he has way more visits than other visitors. I copied his id from session tab. Then I use the heroku cli with the database (postgreSQL in my case), and i am able to run something like this :
```
DELETE FROM session WHERE session_id = 'id_visitor';
DELETE FROM website_event WHERE session_id = 'id_visitor';
```
My question is : will theses DELETE request breaks something or is it fine as there is no other table where I can see the id_visitor?

(I know there is  way to disable umami tracking for a device using dev tools console, but it will be too late and I would like to keep data I already got since the beginning)

**Comment by JorSoi** [comment]
I second this. Deleting specific sessions can be really helpful to keep the analytics clean. The "umami.disabled" option is good but sometimes there is no way around where I want to remove my own sessions or other sessions from QA etc. 

It would also be in the interest of Umami since this is only a problem for the cloud version

**Comment by cabaucom376** [comment]
I find myself also really wanting this since during development I may accidentally spam the analytics with data that is just me testing. The disable flag doesn't do me much good whenever I am testing production websites on mobile with no access to such a thing, or when a non-technical partner is also viewing and testing the website on multiple platforms. I desperately need a better way to delete individual records.

**Comment by Nirator78** [comment]
It can be awesome when some robot are spamming our website too.

**Comment by cabaucom376** [comment]
Just as a heads up, I threw together a PR that implements this. If you guys want to you can build my forks branch.

**Comment by Exonfang** [comment]
+1

While I can manually delete sessions using database commands, having a way to do this directly in GUI saves a lot of headache for day-to-day administration.
